releasenotes/notes/0.4/0.4-summary-3de98711c3b7aa09.yaml

```diff
@@ -0,0 +1,95 @@
+    for details.
+issues:
+  - |
+    JAX release 0.4.4 modified the internals of ``jit`` compilation, resulting in errors when
+    attempting to automatically differentiate and compile calls to :func:`.solve_lmde_perturbation`.
+    This is a bug in JAX, and is discussed in `this discussion thread
+    <https://github.com/google/jax/discussions/14561>`__. Until this is fixed, if using this functionality, JAX can be
+    reverted to the old mode of compilation by executing the following code before importing JAX:
+
+    .. code-block:: python
+
+      import os
+      os.environ["JAX_JIT_PJIT_API_MERGE"] = "0"
+upgrade:
+  - |
+    Starting with this release, Python 3.7 is no longer supported.
+  - |
+    The deprecated ``list`` formatting for the ``carriers`` argument to
+    :class:`.InstructionToSignals` is no longer supported. All code utilizing this argument must be
+    converted to the dictionary input format.
+  - |
+    The deprecated :class:`.Solver` init arguments ``hamiltonian_signals`` and
+    ``dissipator_signals`` have been removed. Signals in the model can now only be passed to the
+    :meth:`.Solver.solve` method.
+  - |
+    The deprecated ``Solver.signals`` property has been removed. The model stored in a
+    :class:`.Solver` instance no longer stores signals.
+  - |
+    The deprecated :meth:`.Solver.copy` method has been removed. There is no replacement for this
+    functionality; the new intended mode of operation for :class:`.Solver` is to be immutable, and
+    hence there is no longer any need to create copies.
+fixes:
+  - |
+    :class:`.InstructionToSignals` has been updated to fix issues with phase accumulation resulting
+    from :class:`~qiskit.pulse.instructions.SetFrequency` and
+    :class:`~qiskit.pulse.instructions.ShiftFrequency` instructions. A phase accumulation term has
+    been added so that the digitized carrier frequency is continuous across these instructions. A
+    related phase error when generating samples for channels subject to
+    :class:`~qiskit.pulse.instructions.SetFrequency` and
+    :class:`~qiskit.pulse.instructions.ShiftFrequency` instructions has also been fixed.
+  - |
+    Fixes a bug in the automatic jit-compilation of :meth:`Solver.solve` when using the ``t_eval``
+    kwarg with a JAX method and ``Array.default_backend() == 'jax'``. The bug is fixed by updating
+    the time-argument handling for the ``"jax_odeint"`` and Diffrax methods. The automatic jitting
+    is now disabled for all other JAX methods.
+  - |
+    The ``jax.Array`` class has been added to the dispatcher for compatibility with JAX 0.4.
+  - |
+    Validation checks have been added to the :class:`.DysonSolver` and :class:`.MagnusSolver`
+    classes to ensure that the ``operators``, ``carrier_freqs``, and ``chebyshev_orders`` arguments
+    have compatible shapes. This fixes bugs in the behaviour of these classes when implicit
+    assumptions about array shapes were violated.
```

Comment:
Are there any issues that were opened for these? If so it might be good to link to the issue so people can look up the context if they're interested.

Reply:
Added links for these.